src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -276,8 +276,10 @@ trait EvalContextPrivExt<'mir: 'ecx, 'tcx: 'mir, 'ecx>: crate::MiriInterpCxExt<'
         let mut tree_borrows = alloc_extra.borrow_tracker_tb().borrow_mut();
         if new_perm.perform_read_access {
             let global = &this.machine.borrow_tracker.as_ref().unwrap();
-            // FIXME: make sure this read counts towards race_condition
             tree_borrows.perform_access(AccessKind::Read, orig_tag, range, global)?;
+            if let Some(data_race) = alloc_extra.data_race.as_ref() {
+                data_race.read(alloc_id, range, &this.machine)?;
+            }
```

Comment:
It seems concerning that no test was updated with this rather fundamental change in behavior... there should be a 'fail' test checking this for SB; can you copy that for TB?